src/sentry/sentry_metrics/indexer/indexer_consumer.py

```diff
@@ -0,0 +1,82 @@
+import logging
+from typing import Any, Dict, Optional, Sequence
+
+from confluent_kafka import Producer
+from django.conf import settings
+
+from sentry.sentry_metrics import indexer
+from sentry.utils import json, kafka_config
+from sentry.utils.batching_kafka_consumer import AbstractBatchWorker, BatchingKafkaConsumer
+from sentry.utils.kafka import create_batching_kafka_consumer
+
+logger = logging.getLogger(__name__)
+
+snuba_metrics = settings.KAFKA_TOPICS[settings.KAFKA_SNUBA_METRICS]
+
+
+def get_metrics_consumer(
+    topic: Optional[str] = None, **options: Dict[str, str]
+) -> BatchingKafkaConsumer:
+    return create_batching_kafka_consumer({topic}, worker=MetricsIndexerWorker(), **options)
+
+
+class MetricsIndexerWorker(AbstractBatchWorker):  # type: ignore
```

Comment:
Please let's have a test case for the worker. This is going to be the real thing, not a mock.